include/simple/server.hpp

```diff
@@ -132,11 +133,11 @@ class Server {
    * callback function and reply.
    */
   void awaitRequest(std::shared_ptr<std::atomic<bool>> alive, std::shared_ptr<GenericSocket<T>> socket) {
```

Comment:
Not directly related to this PR, but still: Why exactly are the two member functions passed again to the thread function as parameters? They should be accessible anyways directly, and passing them only means increasing the refcount. Also, why are they shared_ptr's in the first place? I can see no good reason for it, they should live as long as the thread itself anyways?

Reply:
Ah, might be that this was an issue before, when you did `detach()` instead of `join()` in the destructor. Now it should be guaranteed that the alive_ and socket_ live longer than the thread, so direct access should be possible?

Reply:
Correct! Now that the two threads are joined, we can simplify that function again. The same for the subscriber by the way. Thanks for noticing it